src/main/scala/app/system/Env.scala

```
@@ -1,55 +1,39 @@
 package app.system
 
-import zio.UIO
-import zio.ZIO
-
-sealed trait Env {
-  val raw: String
-}
-
-object Env {
-  case object Prod extends Env {
-    val raw = "prod"
-  }
-
-  case object Dev extends Env {
-    val raw = "dev"
-  }
-
-  case object Test extends Env {
-    val raw = "test"
-  }
-
-  case object Local extends Env {
-    val raw = "local"
-  }
-
-  private def fromRaw(rawEnv: String): Option[Env] = {
-    rawEnv.toLowerCase match {
-      case Prod.raw  => Some(Prod)
-      case Dev.raw   => Some(Dev)
-      case Test.raw  => Some(Test)
-      case Local.raw => Some(Local)
-      case _         => None
-    }
-  }
-
-  val current: UIO[Env] = {
-    val rawEnv = sys.env.get("ENVIRONMENT")
-    ZIO
-      .fromOption(rawEnv.flatMap(fromRaw))
-      .catchAll(_ =>
-        ZIO
-          .logWarning(s"Unknown environment `$rawEnv`. Environment set to `dev`")
-          .as(Dev)
```

I think it's dangerous to default to `Dev`. If you deploy to production without setting the `ENVIRONMENT` envvar (and the service has access to the `Dev` db from there) you risk putting production data(potentially sensitive) into the `Dev` db. It's better to just fail fast and not even boot up.